Jenkinsfile

```diff
@@ -78,130 +77,124 @@ pipeline {
+
+                    environment {
+                        VC = credentials('vcenter')
+                        TESTBED = credentials('win-testbed')
+                        TESTBED_TEMPLATE = "Template-testbed-201804050628"
+                        CONTROLLER_TEMPLATE = "Template-CentOS-7.4-Thin"
+                        TESTENV_MGMT_NETWORK = "VLAN_501_Management"
+                        VC_FOLDER = "WINCI/testenvs"
+                    }
 
-            environment {
-                VC = credentials('vcenter')
-                TESTBED = credentials('win-testbed')
-                TESTBED_TEMPLATE = "Template-testbed-201804050628"
-                CONTROLLER_TEMPLATE = "Template-CentOS-7.4-Thin"
-                TESTENV_MGMT_NETWORK = "VLAN_501_Management"
-                VC_FOLDER = "WINCI/testenvs"
-            }
+                    steps {
+                        script {
+                            def vmwareConfig = getVMwareConfig()
+                            def testNetwork = getLockedNetworkName()
+                            def testEnvName = getTestEnvName(testNetwork)
+                            def testEnvConfig = [
+                                testenv_name: testEnvName,
+                                testenv_vmware_folder: env.VC_FOLDER,
+                                testenv_mgmt_network: env.TESTENV_MGMT_NETWORK,
+                                testenv_data_network: testNetwork,
+                                testenv_testbed_vmware_template: env.TESTBED_TEMPLATE,
+                                testenv_controller_vmware_template: env.CONTROLLER_TEMPLATE
+                            ]
+
+                            def ansibleExtraVars = vmwareConfig + testEnvConfig
 
-            steps {
-                script {
-                    lock(label: 'testenv_pool', quantity: 1) {
-                        def vmwareConfig = getVMwareConfig()
-                        def testNetwork = getLockedNetworkName()
-                        def testEnvName = getTestEnvName(testNetwork)
-                        def testEnvConfig = [
-                            testenv_name: testEnvName,
-                            testenv_vmware_folder: env.VC_FOLDER,
-                            testenv_mgmt_network: env.TESTENV_MGMT_NETWORK,
-                            testenv_data_network: testNetwork,
-                            testenv_testbed_vmware_template: env.TESTBED_TEMPLATE,
-                            testenv_controller_vmware_template: env.CONTROLLER_TEMPLATE
-                        ]
-
-                        ansibleExtraVars = vmwareConfig + testEnvConfig
-
-                        // 'Cleanup' stage
-                        node(label: 'ansible') {
                             deleteDir()
                             unstash 'Ansible'
 
                             dir('ansible') {
+                                // Cleanup testenv before making a new one
```

Comment:
This makes me wonder...

Originally, we wanted to destroy testenv in `post` stage. We moved it earlier because of issues with holding a lock for so long.

In this proposed PR, lock is pretty much global. Does this mean we can move testenv destruction back into `post` stage? Do we want to? Depending on how long testenv provisioning takes compared to build - it might be beneficial.

Reply:
I think it should be doable. However it would include some refactor since we would need `ansibleExtraVars`  also in `post` stage. Moreover I think it might be a bit risky if the job would assume that testenv is already cleared. I don't want to do that in this PR. However yes it should be beneficial.